internal/provider/indices_resource.go

```diff
@@ -839,9 +875,117 @@ func (r *indicesResource) Create(ctx context.Context, req resource.CreateRequest
 		return
 	}
 
-	// Set the index name as the ID in the Terraform state
+	// Set initial state
+	model.MarqoEndpoint = types.StringValue("pending")
 	diags = resp.State.Set(ctx, &model)
 	resp.Diagnostics.Append(diags...)
+
+	// Wait for index to be ready
+	timeout := time.After(timeoutDuration)
+	ticker := time.NewTicker(30 * time.Second)
+	defer ticker.Stop()
+
+	tflog.Info(ctx, fmt.Sprintf("Waiting up to %v for index %s to be ready...",
+		timeoutDuration,
+		model.IndexName.ValueString()))
+
+	start := time.Now()
+	for {
+		select {
+		case <-timeout:
+			// Check the current status of the index
+			indices, err := r.marqoClient.ListIndices()
+			if err != nil {
+				resp.Diagnostics.AddError(
+					"Failed to Check Index Status",
+					fmt.Sprintf("Could not check index status after timeout: %s", err),
+				)
+				return
+			}
+
+			var indexStatus string
+			for _, index := range indices {
+				if index.IndexName == model.IndexName.ValueString() {
+					indexStatus = index.IndexStatus
+					break
+				}
+			}
+
+			if indexStatus == "CREATING" {
+				resp.Diagnostics.AddError(
+					"Timeout Waiting for Index",
+					fmt.Sprintf("Index %s did not become ready within the %v timeout period. "+
+						"The index is still being created in the cloud and cannot be deleted at this time. "+
+						"You may need to manually delete it later using the Marqo console or API.",
+						model.IndexName.ValueString(),
+						timeoutDuration),
+				)
+				return
+			}
+
+			// If the index is in any other state, try to delete it
+			deleteErr := r.marqoClient.DeleteIndex(model.IndexName.ValueString())
+			if deleteErr != nil {
+				resp.Diagnostics.AddError(
+					"Cleanup Failed",
+					fmt.Sprintf("Index %s creation timed out after %v and cleanup failed: %s. "+
+						"Manual cleanup may be required.",
+						model.IndexName.ValueString(),
+						timeoutDuration,
+						deleteErr),
+				)
+				return
+			}
+
+			resp.Diagnostics.AddError(
+				"Timeout Waiting for Index",
+				fmt.Sprintf("Index %s did not become ready within the %v timeout period and has been deleted.",
+					model.IndexName.ValueString(),
+					timeoutDuration),
+			)
+			return
+		case <-ticker.C:
+			indices, err := r.marqoClient.ListIndices()
+			if err != nil {
+				tflog.Error(ctx, fmt.Sprintf("Error listing indices: %s", err))
+				continue
+			}
+			for _, index := range indices {
+				if index.IndexName == model.IndexName.ValueString() {
+					tflog.Info(ctx, fmt.Sprintf("Index %s status: %s (elapsed: %v)",
+						model.IndexName.ValueString(),
+						index.IndexStatus,
+						time.Since(start)))
+					if index.IndexStatus == "READY" {
+						tflog.Info(ctx, fmt.Sprintf("Index %s is now ready (total time: %v)",
+							model.IndexName.ValueString(),
+							time.Since(start)))
+
+						// Do final read to get the complete state
+						readResp := resource.ReadResponse{
+							State: resp.State,
+						}
```

Comment:
Nit: can these go on one line (this and others)? Newlines for a single field seem overkill.
```suggestion
						readResp := resource.ReadResponse{State: resp.State}
```